### PR TITLE
fix(gateway): fix topology response

### LINKED ISSUE #2357: Topology contains wrong partition information for node
Since the migration to atomix we observed some weird behavior on the requested topology. It seems that each node is follower for each partition, even if the replication factor is 1 and cluster size is 3.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
* topology contains correct number of follower and leader per node

closes #2357

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java [hunk 1]
```diff
@@ -242,9 +242,13 @@ private BrokerInfo createDistributionTopology() {
...
       final int partitionId = partitionInfo.getPartitionId();
       final NodeInfo leader = topology.getLeader(partitionId);
+      final List<NodeInfo> followers = topology.getFollowers(partitionId);
 
       final boolean isLeader = leader != null && leader.getNodeId() == local.getNodeId();
-      distributionInfo.setPartitionRole(partitionId, isLeader);
+
+      if (isLeader || followers.contains(local)) {
+        distributionInfo.setPartitionRole(partitionId, isLeader);
+      }
```
Comment (deepthidevaki) [inline broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java]: Wouldn't this be sufficient here?
```    
for (PartitionInfo partitionInfo : local.getLeaders()) {
      final int partitionId = partitionInfo.getPartitionId();
        distributionInfo.setPartitionRole(partitionId, true);
    }
for (PartitionInfo partitionInfo : local.getFollowers()) {
      final int partitionId = partitionInfo.getPartitionId();
      distributionInfo.setPartitionRole(partitionId, false);
    }
```

**Review by deepthidevaki — COMMENTED** [review]
Looks good. :+1: Only a minor comment.

**Comment by Zelldon** [comment]
@deepthidevaki you're right I changed that and did some more refactoring, please have a look :)

**Comment by deepthidevaki** [comment]
:+1:

**Review by deepthidevaki — APPROVED** [review]

**Pull request merged**